CalibCalorimetry/HcalAlgos/src/HcalDbHardcode.cc

```diff
@@ -564,30 +564,37 @@ HcalSiPMParameter HcalDbHardcode::makeHardcodeSiPMParameter (HcalGenericDetId fI
   HcalSiPMType theType = HcalNoSiPM;
   double thePe2fC = getParameters(fId).photoelectronsToAnalog();
   double theDC = getParameters(fId).darkCurrent(0);
-  if (fId.genericSubdet() == HcalGenericDetId::HcalGenBarrel && useHBUpgrade_) {
-    HcalDetId hid(fId);
-    int nLayersInDepth = getLayersInDepth(hid.ietaAbs(),hid.depth(),topo);
-    if(nLayersInDepth > 4) {
-      theType = HcalHBHamamatsu2;
-      theDC = getParameters(fId).darkCurrent(1);
-    }
-    else {
-      theType = HcalHBHamamatsu1;
-      theDC = getParameters(fId).darkCurrent(0);
-    }
-  } else if (fId.genericSubdet() == HcalGenericDetId::HcalGenEndcap && useHEUpgrade_) {
-    HcalDetId hid(fId);
-    int nLayersInDepth = getLayersInDepth(hid.ietaAbs(),hid.depth(),topo);
-    if(nLayersInDepth > 4) {
-      theType = HcalHEHamamatsu2;
-      theDC = getParameters(fId).darkCurrent(1);
+  if (fId.genericSubdet() == HcalGenericDetId::HcalGenBarrel) {
+    if(useHBUpgrade_) {
+      HcalDetId hid(fId);
+      int nLayersInDepth = getLayersInDepth(hid.ietaAbs(),hid.depth(),topo);
+      if(nLayersInDepth > 4) {
+        theType = HcalHBHamamatsu2;
+        theDC = getParameters(fId).darkCurrent(1);
+      }
+      else {
+        theType = HcalHBHamamatsu1;
+        theDC = getParameters(fId).darkCurrent(0);
+      }
     }
-    else {
-      theType = HcalHEHamamatsu1;
-      theDC = getParameters(fId).darkCurrent(0);
+    else theType = HcalHPD;
+  } else if (fId.genericSubdet() == HcalGenericDetId::HcalGenEndcap) {
+    if(useHEUpgrade_) {
+      HcalDetId hid(fId);
+      int nLayersInDepth = getLayersInDepth(hid.ietaAbs(),hid.depth(),topo);
+      if(nLayersInDepth > 4) {
+        theType = HcalHEHamamatsu2;
+        theDC = getParameters(fId).darkCurrent(1);
+      }
+      else {
+        theType = HcalHEHamamatsu1;
+        theDC = getParameters(fId).darkCurrent(0);
+      }
     }
-  } else if (fId.genericSubdet() == HcalGenericDetId::HcalGenOuter && useHOUpgrade_) {
-    theType = HcalHOHamamatsu;
+    else theType = HcalHPD;
+  } else if (fId.genericSubdet() == HcalGenericDetId::HcalGenOuter) {
+    if(useHOUpgrade_) theType = HcalHOHamamatsu;
+    else theType = HcalHPD;
   }
```

Comment:
Hi @kpedro88 
sorry for coming to this only now.
I try to understand the change of the logic here:
It looks to me as if the new enum type is used in case of `useHOUpgrade == false` or `useHEUpgrade == false`, i.e. it is only valid for past runs?
Where is this type then used and what purpose does it serve?